js/src/framework.ts

```diff
@@ -393,23 +490,59 @@ export async function runEvaluator(
                   if (scoreValue === null) {
                     return null;
                   }
-                  const result: Score =
+
+                  const scoreValues = Array.isArray(scoreValue)
+                    ? scoreValue
+                    : [scoreValue];
+
+                  const results: Score[] = scoreValues.map((scoreValue, idx) =>
                     typeof scoreValue === "object"
                       ? scoreValue
-                      : { name: scorerNames[score_idx], score: scoreValue };
-                  const {
-                    metadata: resultMetadata,
-                    name,
-                    ...resultRest
-                  } = result;
+                      : {
+                          name:
+                            scoreValues.length > 1
+                              ? `${scorerNames[score_idx]}_${idx}`
+                              : scorerNames[score_idx],
+                          score: scoreValue,
+                        }
+                  );
+
+                  const getOtherFields = (s: Score) => {
+                    const { metadata, name, ...rest } = s;
+                    return rest;
+                  };
+
+                  const resultMetadata =
+                    results.length === 1
```

Comment:
nit: the branching seems unnecessary here? Not worth optimizing I feel.

Reply:
i did this because it has a non trivial impact on our expect tests and the UX -- basically did not want to regress the experience for existing (common) case of returning 1 score.

Reply:
oh yes I see that the reduce changes it to `{ [s.name]: s.metadata }` instead of just the metadata. so never mind.